pkg/cmd/grafana-server/server.go

```diff
@@ -72,6 +77,7 @@ func (g *GrafanaServerImpl) Run() error {
 	serviceGraph.Provide(&inject.Object{Value: bus.GetBus()})
 	serviceGraph.Provide(&inject.Object{Value: g.cfg})
 	serviceGraph.Provide(&inject.Object{Value: routing.NewRouteRegister(middleware.RequestMetrics, middleware.RequestTracing)})
+	serviceGraph.Provide(&inject.Object{Value: cache.New(5*time.Minute, 10*time.Minute)})
```

Comment:
The reason for this being a service is that OSS code and GE code want to share the same cache?

Otherwise, I don't think caching is a service that should be shared within Grafana. Each consumer of a cache should be able to have different duration and cache invalidation strategy depending on how the data should be cached. This would also make all cached values accessible for all services which might cause problems due to bad cache keys.

Reply:
This was added as a replacement of [existing cache for http server](https://github.com/grafana/grafana/blob/master/pkg/api/http_server.go#L61). Each consumer of the cache decide for how long an item should be cached ([ref](https://github.com/grafana/grafana/blob/master/pkg/api/dataproxy.go#L49)).

> This would also make all cached values accessible for all services which might cause problems due to bad cache keys.

I agree that this might not be good. However using memcache or redis you would always have the same issue of bad cache keys. I was also afraid of that having multiple instances of go-cache would add some overhead plus there's currently only one consumer of the http server cache instance in the code base. 

We can easily change this so that each service that needs caching instantiate their own instance of the cache.

Reply:
I think a shared cache service make 100% sense, this is something that I have been wanting to do for a long time since we added the service infrastructure. 

Every consumer can specify the expiration time, the one added here is just the default.

Reply:
I changed my mind in this case. We need something like this to enable caching within GE.

But I'm still not totally satisfied with injecting all deps with current implementation but I think I can come up with some suggestion for how to make it better.

Reply:
We can totally refactor this later. Will merge this now.